src/main/host/syscall/ioctl.c

```diff
@@ -173,6 +174,32 @@ static int _syscallhandler_ioctlUDPHelper(SysCallHandler* sys, UDP* udp, int fd,
     return result;
 }
 
+static int _syscallhandler_ioctlEventDHelper(SysCallHandler* sys, EventD* eventd, int fd,
+                                             unsigned long request, PluginPtr argPtr) {
+    int result = -EINVAL;
+
+    switch (request) {
+        case FIONBIO: {
+            const int* val = process_getReadablePtr(sys->process, argPtr, sizeof(int));
```

Comment:
Nit - for a small value like an int, `process_readPtr`  is preferable. It'll be faster in the case that the MM goes through the `readv` path, since it doesn't need to allocate and deallocate a buffer. It should be no slower in the case that the MM goes through the mapped path, since the cost to copy the int is negligible.

Reply:
Remember to check for a NULL return value (or an error indicator from `process_readPtr`)

Reply:
I updated all of the ioctl syscall handlers to use process_readPtr and process_writePrt.